tortoise_interpreter.py

```diff
@@ -20,41 +20,54 @@ def sanitize_program(program):
 
 
 def navigate_command_functions(COMMANDS, program):
-    # Comment
-    return [f'{i}. {COMMANDS[command[0]](command)}' for i, command in enumerate(program, 1)]
+    command_to_parse = []
+    pen_state = ''
+    for i, command in enumerate(program, 1):
+        if command[0] in ['U', 'D']:
+            pen_state = get_pen_state(command)
```

Comment:
I don't think that the `navigate_command_functions` function should need to know which commands alter the state. The Up/Down functions should probably do that themselves.